### PR TITLE
Convert ReusableTextReader to be a struct

### PR DESCRIPTION
**Pull request by jamesqo**
Since the `ReusableTextReader` class in Common is sealed with all-readonly fields, it looks like it could be converted to a struct (the field references are copied, not the contents themselves). I've changed the type declaration to be a struct in this PR, as well as modifying all call sites that used the parameterless constructor.

Here is a list of all the places it's being used currently (all in System.Diagnostics.Process): https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=ReusableTextReader

cc @stephentoub

### CODE REVIEW

File: src/Common/src/System/Text/ReusableTextReader.cs [hunk 1]
```diff
@@ -7,10 +7,10 @@
 namespace System.Text
 {
     /// <summary>Provides a reusable reader for reading all of the text from streams.</summary>
-    internal sealed class ReusableTextReader
+    internal struct ReusableTextReader
```
Comment (hughbe) [inline src/Common/src/System/Text/ReusableTextReader.cs]: If the struct is only used in System.Diagnostics.Process, why is it in comon?

**Comment by jamesqo** [comment]
Whoops, sorry for the failing CI tests. That's strange, I thought I'd tested it locally. Will take a look sometime in the next day or two to see why that's happening.

**Comment by stephentoub** [comment]
> That's strange, I thought I'd tested it locally.

It doesn't compile.  Closing for now.

File: src/Common/src/System/Text/ReusableTextReader.cs (hunk 1 above)
Comment (stephentoub) [inline src/Common/src/System/Text/ReusableTextReader.cs]: It's a general purpose thing that is likely to be used elsewhere.

**Comment by jamesqo** [comment]
@stephentoub My apologies, I'd forgotten to add `/p:OSGroup=Linux` when testing locally. I've pushed a new commit letting it build, could you PTAL at reopening this?

**Comment by jamesqo** [comment]
Will close temporarily again due to the failures, apologies since I'm not sure why this is happening.

**Comment by stephentoub** [comment]
I'm not convinced this is an overall beneficial change.  The majority use case for this type is creating it once and passing it around to be used by a whole bunch of users, e.g. using it to parse the statfs files for all current processes.  Creating one of these already involves allocating several large data structures, e.g. a char[], a byte[], a decoder, etc., and as a class it's adding only one additional small one.  In contrast, as a struct, every time we pass it around, we're copying multiple fields.  Without a benchmark to show that it meaningfully moves the needle in a particular direction, I'm inclined to leave things the way they are and avoid the churn.  Thanks, though.

**Pull request closed without merging**